lib/main.dart

```diff
@@ -1,123 +1,75 @@
-final currentUri = Uri.base;
-final redirectUri = Uri(
-  host: currentUri.host,
-  scheme: currentUri.scheme,
-  port: currentUri.port,
-  path: '/index.html',
-);
-final authUrl =
-    'https://cloud.ouraring.com/oauth/authorize?client_id=$clientId&state=$clientSecret&redirect_uri=$redirectUri&response_type=token';
+import 'login_widget.dart';
+import 'loading_widget.dart';
 
 void main() async {
-  runApp(const MyApp());
+  runApp(const AppGlobalState());
 }
 
-///
-Future<String> _login(String data, popupWindow) async {
-  /// Parse data into an Uri to extract the token easily.
-  final receivedUri = Uri.parse(data);
-
-  /// Parse fragment string to token
-  var token = receivedUri.fragment
-      .split('&')
-      .firstWhere((e) => e.startsWith('access_token='))
-      .substring('access_token='.length);
-
-  if (popupWindow != null) {
-    popupWindow.close();
-    popupWindow = null;
-  }
+class AppGlobalState extends StatefulWidget {
+  const AppGlobalState({Key? key}) : super(key: key);
 
-  return token;
+  @override
+  State<AppGlobalState> createState() => _AppGlobalStateState();
 }
 
-class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
-
+class _AppGlobalStateState extends State<AppGlobalState> {
+  // User login state.
+  String? _userToken;
+  String? _userSecret;
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      initialRoute: _userToken == null ? '/login' : '/dashboard',
+      routes: {
+        '/login': (context) => LoginWidget(
+              onLogin: (identity, secret) {
+                setState(() {
+                  _userToken = identity;
+                  _userSecret = secret;
+                });
+              },
+            ),
+        '/dashboard': (context) => Column(
+              children: [
+                Text('..Hello, $_userToken...'),
+                Text(
+                  'Please.. Make a pull request that changes me into a real dashboard...',
+                  textScaleFactor: 0.8,
+                ),
+                LoadingWidget(),
+                Text(
+                  'Ill be loading in the meanwhile...',
+                  textScaleFactor: 0.3,
+                ),
+                Text(
+                  'Your secret is $_userSecret, please keep it safe. :)',
+                  textScaleFactor: 0.6,
+                ),
+                Text(
+                  'Never print it in big red text, it is a secret! Remember that.',
+                  textScaleFactor: 0.8,
+                ),
+              ],
+            ),
+        '/settings': (context) =>
+            throw UnimplementedError("TODO place ex. SettingsWidget here"),
+        '/settings/addservice': (context) =>
+            throw UnimplementedError("TODO place ex. ServiceListWidget here"),
+        '/services/oura': (context) =>
+            throw UnimplementedError("TODO place Jakob's oura auth page here"),
```

Comment:
Not sure what this page is supposed to be since the oura login page is implemented externally.